tests/int_tests.py

```diff
@@ -8,20 +9,46 @@
 from common import OdmTestCase, readfile
 
 
-class IntTests(OdmTestCase):
-    def test_excel_delatolla(self) -> None:
-        delatolla_dir = join(self.dir, 'int', 'delatolla')
-        schema_path = join(delatolla_dir, 'schema.csv')
-        data_path = join(delatolla_dir, 'data.xlsx')
+def _remove_file(path):
+    try:
+        os.remove(path)
+    except FileNotFoundError:
+        pass
+
+
+class DelatollaIntTests(OdmTestCase):
+    def setUp(self):
+        super().setUp()
+        self.delatolla_dir = join(self.dir, 'int', 'delatolla')
+        self.dbname = 'data.sqlite'
+        self.dbpath = join(self.delatolla_dir, self.dbname)
+
+        # remove db from previous run
+        _remove_file(self.dbpath)
+
+    def _test_impl(self, data_filename) -> None:
+        data_path = join(self.delatolla_dir, data_filename)
+        schema_path = join(self.delatolla_dir, 'schema.csv')
         with TemporaryDirectory() as tmpdir:
             paths = share(schema_path, data_path, outdir=tmpdir,
                           outfmt=OutFmt.CSV)
             for path in paths:
                 actual = readfile(path)
                 fn = Path(path).name
-                expected = readfile(join(delatolla_dir, 'expected', fn))
+                expected = readfile(join(self.delatolla_dir, 'expected', fn))
                 self.assertEqual(actual, expected)
 
+    def test(self) -> None:
+
+        # test excel
+        os.environ['ODM_TEMP_DB'] = self.dbpath
```

Comment:
Rather than using this environment variable and modifying the function, can we not create a sqlite file? That way the testing logic is not leaking into the business logic.

Reply:
well, it's a nice env var to have because i find myself changing the db path all the time during development (since i want to be able to inspect the intermediary data), so it's not just used by the tests but also the developers (and users if they want). If we add the sqlite file to the repo, then we have to remember to update it every time we update the xlsx file, and it's also a relatively big binary file which would slow down and increase the repo size for limited benefit.